src/main/ruby/core/dir_glob.rb

```diff
@@ -29,6 +29,7 @@
 class Dir
   module Glob
     NO_GLOB_META_CHARS = /^[^*?\[\]{}\\]+$/
+    TRAILING_BRACES = /^([^*?\[\]{}\\]+)(?:\{([^*?\[\]{}\\]*)\})?$/
```

Comment:
It would be nice to share the common sub-expressions, i.e., the set of meta-chars, also with the Regexp above, for instance by using a local variable.